TAO/TAO_IDL/driver/drv_preproc.cpp

```diff
@@ -577,11 +578,107 @@ DRV_add_include_path (ACE_CString& include_path,
   return include_path;
 }
 
+namespace {
+  const char *our_basename(const char *path)
+  {
+    const char *temp;
+    while ((temp = ACE_OS::strchr (path, ACE_DIRECTORY_SEPARATOR_CHAR_A)))
+      {
+        path = temp + 1;
+      }
+    return path;
+  }
+
+  char *tolower(const char *input)
+  {
+    if (!input) return nullptr;
+    const size_t size = ACE_OS::strlen (input);
+    char *output = nullptr;
+    ACE_NEW_RETURN (output, char [size], nullptr);
+    for (size_t i = 0; i < size; ++i) output[i] = ACE_OS::ace_tolower(input[i]);
+    return output;
+  }
+
+  /// Copy IDL input file for preprocessor?
+  bool copy = true;
+}
+
 // Adds additional include paths, but after parse_args() has
 // added user-defined include paths.
 void
 DRV_cpp_post_init ()
 {
+  char idl_version_arg[128];
+  ACE_OS::sprintf (idl_version_arg, "-D__TAO_IDL_IDL_VERSION=%s",
+    idl_global->idl_version_.to_macro ());
+  DRV_cpp_putarg (idl_version_arg);
+
+  // Decide how files will be passed to the preprocessor
+  if (idl_global->preprocessor_input_ == IDL_GlobalData::PreprocessorInputGuess)
+    {
+      const char * const cpp_name = tolower (our_basename (FE_get_cpp_loc_from_env ()));
```

Comment:
What about setting the default through the various config-* files, there we also set the default preprocessor to be used?